lib/datadog/appsec/contrib/rack/request_body_middleware.rb

```diff
@@ -0,0 +1,41 @@
+# typed: ignore
+
+require 'datadog/appsec/instrumentation/gateway'
+require 'datadog/appsec/assets'
+
+module Datadog
+  module AppSec
+    module Contrib
+      module Rack
+        # Rack request body middleware for AppSec
+        # This should be inserted just below Rack::JSONBodyParser or
+        # legacy Rack::PostBodyContentTypeParser from rack-contrib
+        class RequestBodyMiddleware
+          def initialize(app, opt = {})
+            @app = app
+          end
+
+          def call(env)
+            context = env['datadog.waf.context']
+
+            return @app.call(env) unless context
+
+            # TODO: handle exceptions, except for @app.call
```

Comment:
Is this todo for the future, or for this pr?

Reply:
Future, it's a tough nut to crack.